### PR TITLE
TYP: read_csv's usecols

### PR DESCRIPTION
**Pull request by twoertwein**
This makes read_csv's usecols compatible with the tests from pandas-stubs.

The documentation does mention only list, Callable, and None! This notably excludes range, tuple, and ndarray. I'm happy to tighten the types to the documented types or adjust the documentation.

edit: The pandas tests cover list, tuple, set, nd.array and probably more -> I adjusted the documentation

### CODE REVIEW

File: pandas/_typing.py [hunk 1]
```diff
@@ -507,3 +507,12 @@ def closed(self) -> bool:
 
 # Offsets
 OffsetCalendar = Union[np.busdaycalendar, "AbstractHolidayCalendar"]
+
+# read_csv: usecols
+UsecolsArgType = Union[
+    SequenceNotStr[Hashable],
+    Sequence[int],
```
Comment (twoertwein) [inline pandas/_typing.py:514]: needed for `range`: `range` inherits from `Sequence` but changes some of the signatures so it doesn't fit the `SequenceNotStr` protocol

Comment (Dr-Irv) [reply pandas/_typing.py:516]: Why not just be explicit with `range` ?

Comment (twoertwein) [reply pandas/_typing.py]: changed it to range

**Review by Dr-Irv — APPROVED** [review]
thanks @twoertwein .  There are still slight differences between `pandas-stubs` and `pandas` here.  Do you want to create an issue or PR for the stubs to align things?

**Pull request merged**